plugins/filter/list.py

```diff
@@ -1,43 +1,88 @@
 # -*- coding: utf-8 -*-
-# Copyright (c) 2020, Vladimir Botka <vbotka@gmail.com>
+# Copyright (c) 2020-2022, Vladimir Botka <vbotka@gmail.com>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
-from ansible.errors import AnsibleError, AnsibleFilterError
+from ansible.errors import AnsibleFilterError
 from ansible.module_utils.six import string_types
 from ansible.module_utils.common._collections_compat import Mapping, Sequence
+from ansible.utils.vars import merge_hash
 from collections import defaultdict
 from operator import itemgetter
 
 
-def lists_mergeby(l1, l2, index):
-    ''' merge lists by attribute index. Example:
-        - debug: msg="{{ l1|community.general.lists_mergeby(l2, 'index')|list }}" '''
-
-    if not isinstance(l1, Sequence):
-        raise AnsibleFilterError('First argument for community.general.lists_mergeby must be list. %s is %s' %
-                                 (l1, type(l1)))
-
-    if not isinstance(l2, Sequence):
-        raise AnsibleFilterError('Second argument for community.general.lists_mergeby must be list. %s is %s' %
-                                 (l2, type(l2)))
-
-    if not isinstance(index, string_types):
-        raise AnsibleFilterError('Third argument for community.general.lists_mergeby must be string. %s is %s' %
-                                 (index, type(index)))
+def list_mergeby(x, y, index, recursive=True, list_merge='replace'):
+    ''' Merge 2 lists by attribute 'index'. The function merge_hash from ansible.utils.vars is used.
+        This function is used by the function lists_mergeby. '''
 
     d = defaultdict(dict)
-    for l in (l1, l2):
+    for l in (x, y):
         for elem in l:
             if not isinstance(elem, Mapping):
                 raise AnsibleFilterError('Elements of list arguments for lists_mergeby must be dictionaries. Found {0!r}.'.format(elem))
             if index in elem.keys():
-                d[elem[index]].update(elem)
+                d[elem[index]].update(merge_hash(d[elem[index]], elem, recursive, list_merge))
```

Comment:
Thank you for the comments! I created a wrapper to deal with it. It's not such a burden and might be useful in the future. What do you think?

I'll proceed with the changelog and doc. In the doc, my intention is to refer to [Combining hashes/dictionaries](https://docs.ansible.com/ansible/devel/user_guide/playbooks_filters.html#combining-hashes-dictionaries) because the function 'merge_hash' from ansible.utils.vars, used by the filter 'combine', is also used by the filter 'lists_mergeby'. Are there any rules for such cross-referencing?

Reply:
There are no rules for cross-referencing yet.